kernel/src/checkpoint/log_replay.rs

```diff
@@ -0,0 +1,603 @@
+//! The [`CheckpointLogReplayProcessor`] implements specialized log replay logic for creating
+//! checkpoint files. It processes log files in reverse chronological order (newest to oldest)
+//! and selects only the minimal set of actions needed to represent the table state at a given version.
+//!
+//! ## Filtering Process
+//!
+//! For checkpoint creation, this processor applies several filtering and deduplication
+//! steps to each batch of log actions:
+//!
+//! 1. **Protocol and Metadata**: Retains only the latest protocol and metadata actions.
+//! 2. **Transactions**: Keeps the most recent action for each unique transaction (by app ID).
+//! 3. **File Actions**: Deduplicates file actions (add/remove) by path and deletion vector ID,
```

Comment:
can we say something along the lines of "resolves file actions to produce the latest state of adds for the table"